cloud_storage_handler/main.py

```diff
@@ -39,13 +32,56 @@ def init_app() -> FlaskApp:
 
     foca = Foca(
         config_file=config_path,
+        custom_config_model="cloud_storage_handler.custom_config.CustomConfig",
     )
-    return foca.create_app()
+    foca_config = foca.conf
+    try:
+        custom_config_data = foca_config.custom.dict()
+        custom_config = CustomConfig(**custom_config_data)
+    except ValidationError as e:
+        logger.error(f"Error parsing custom configuration: {e}")
+        raise
+    minio_config = custom_config.minio
+
+    # Create the Connexion FlaskApp instance
+    return foca.create_app(), minio_config
 
 
 def main() -> None:
     """Run FOCA application."""
-    app = init_app()
+    try:
+        app, minio_config = init_app()
+    except Exception as e:
+        logger.error(f"Unexpected error during initialization: {e}")
+        sys.exit(1)
+    foca_app = app.app
+
+    # Initialize MinIO client
```

Comment:
The code below should pretty much go into `client.MinioClient`, which should have a constructor for creating a client instance, and for now a method `.create_bucket()` - pretty much like you had previously. Just we want to have just a single module, not a package and multiple modules.

Reply:
After reading all of the code, perhaps we really don't need `client.MinioClient` - because `minio.Minio` already has pretty much everything we will need (and more). Sure, having `.create_bucket()` instead of using `.make_bucket()` with `bucket_exists()` is a handy utility function. But until we use this at least in three different places, I think we can live with the additional line it takes to do the check manually. Or we just define a `utils.py` module and add client helper functions like `create_bucket()` there.